eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/tcp/client/group/ClientSessionGroupMapping.java

```diff
@@ -325,60 +320,57 @@ private void handleUnackMsgsInSession(Session session) {
     }
 
-    private void shutdownClientGroupConsumer(Session session) throws Exception {
-        if (Objects.requireNonNull(session.getClientGroupWrapper().get()).started4Broadcast.get() == Boolean.TRUE) {
-            Objects.requireNonNull(session.getClientGroupWrapper().get()).shutdownBroadCastConsumer();
+    private void shutdownClientGroupConsumer(ClientGroupWrapper clientGroupWrapper) throws Exception {
+        if (clientGroupWrapper.started4Broadcast.get() == Boolean.TRUE) {
+            clientGroupWrapper.shutdownBroadCastConsumer();
         }
 
-        if (Objects.requireNonNull(session.getClientGroupWrapper().get()).started4Persistent.get() == Boolean.TRUE) {
-            Objects.requireNonNull(session.getClientGroupWrapper().get()).shutdownPersistentConsumer();
+        if (clientGroupWrapper.started4Persistent.get() == Boolean.TRUE) {
+            clientGroupWrapper.shutdownPersistentConsumer();
         }
     }
 
 
-    private void shutdownClientGroupProducer(Session session) throws Exception {
-        if (Objects.requireNonNull(session.getClientGroupWrapper().get()).producerStarted.get() == Boolean.TRUE) {
-            Objects.requireNonNull(session.getClientGroupWrapper().get()).shutdownProducer();
+    private void shutdownClientGroupProducer(ClientGroupWrapper clientGroupWrapper) throws Exception {
+        if (clientGroupWrapper.producerStarted.get() == Boolean.TRUE) {
+            clientGroupWrapper.shutdownProducer();
         }
     }
 
     private void initSessionCleaner() {
         eventMeshTCPServer.getScheduler().scheduleAtFixedRate(
-                new Runnable() {
-                    @Override
-                    public void run() {
-                        for (Session tmp : sessionTable.values()) {
-                            if (System.currentTimeMillis() - tmp.getLastHeartbeatTime()
-                                > eventMeshTCPServer.getEventMeshTCPConfiguration().eventMeshTcpSessionExpiredInMills) {
-                                try {
-                                    log.warn("clean expired session,client:{}", tmp.getClient());
-                                    closeSession(tmp.getContext());
-                                } catch (Exception e) {
-                                    log.error("say goodbye to session error! {}", tmp, e);
-                                }
-                            }
+            () -> {
+                for (Session tmp : sessionTable.values()) {
+                    if (System.currentTimeMillis() - tmp.getLastHeartbeatTime()
+                        > eventMeshTCPServer.getEventMeshTCPConfiguration().eventMeshTcpSessionExpiredInMills) {
+                        try {
+                            log.warn("clean expired session,client:{}", tmp.getClient());
+                            closeSession(tmp.getContext());
+                        } catch (Exception e) {
+                            log.error("say goodbye to session error! {}", tmp, e);
```

Comment:
suggest using String.format to format log message.

Reply:
Why use string.format instead of just using the placeholders provided by log? Many optimization issues have been changed to use llog's placeholder approach

Reply:
public void error(String format, Object... arguments);
  vs
public void error(String msg, Throwable t);

if you want to use public void error(String format, Object... arguments)，but  variable num is invalid. @weihubeats

Reply:
What is the variable num?
What's wrong with this log method?
<img width="1219" alt="image" src="https://user-images.githubusercontent.com/42484192/210234227-6ae5985c-121a-4828-a1de-0c10009a54d9.png">
<img width="1178" alt="image" src="https://user-images.githubusercontent.com/42484192/210234439-c78d2a80-473d-472d-a07b-fee8e99eaa6b.png">